pkg/config/nodetreemodel/config.go

```diff
@@ -296,6 +298,24 @@ func (c *ntmConfig) mergeAllLayers() error {
 	return nil
 }
 
+func computeAllSettings(node InnerNode, path string) []string {
+	knownKeys := []string{}
+	for _, name := range node.ChildrenKeys() {
+		newPath := name
+		if path != "" {
+			newPath = joinKey(path, name)
+		}
+
+		child, _ := node.GetChild(name)
+		if _, ok := child.(LeafNode); ok {
+			knownKeys = append(knownKeys, newPath)
+		} else {
```

Comment:
```suggestion
		} else if inner, ok := child.(InnerNode); ok {
```
then use `inner` below, and add an else case that should log.Error. That is more explicit and will help in case there's ever a new node type that we need to add down the line, or if `GetChild` returns something weird.